src/plugins/invisibleChat/index.tsx

```diff
@@ -213,7 +227,31 @@ export function encrypt(secret: string, password: string, cover: string): string
     return steggo.hide(secret + "\u200b", password, cover);
 }
 
-export function decrypt(secret: string, password: string): string {
-    return steggo.reveal(secret, password).replace("\u200b", "");
+export function decrypt(secret: string, password: string, removeIndicator: boolean): string {
+    const decrypted = steggo.reveal(secret, password);
+    return removeIndicator ? decrypted.replace("\u200b", "") : decrypted;
 }
 
+export function isCorrectPassword(result: string): boolean {
+    return result.endsWith("\u200b");
+}
+
+export async function iteratePasswords(message: Message): Promise<string | false> {
+    const passwords = settings.store.savedPasswords.split(", ");
+
+    if (!message || !message?.content || !passwords || !passwords.length) return false;
+
+    let { content } = message;
+
+    // we use an extra variable so we dont have to edit the message content directly
+    if (message.content.match(/^\W/)) content = `d ${message.content}d`;
```

Comment:
wait whats going on with these `d`s? Also regex.test is the way to go when you want a boolean!
```suggestion
    if (/^\W/.test(message.content)) content = `d ${message.content}d`;
```

Reply:
The d's simulate a normal message with 2 words. This is necessary to avoid stegcloak mangling the decrypted message